src/main/kotlin/com/target/slack/EmojiService.kt

```diff
@@ -335,18 +336,71 @@ class EmojiService(private val config: Config, private val db: Db) {
                     )
                 )
                 valid = true
+
+                val tallyResult = tallyVotes(prop)
+                // Should we report this to the admins?
+                if (tallyResult.systemReport == 0 && tallyResult.down >= config.votes.downVoteThreshold) {
+                    logger.info("Proposal ${prop.id} has ${tallyResult.down} downvotes. Attempting to flag.")
+                    val r = ctx.client().reactionsAdd() {
+                        it.channel(config.slack.slackEmojiChannel)
+                        it.timestamp(ts)
+                        it.name(Proposals.REPORT)
+                    }
+
+                    if (r.isOk) {
+                        recordVote(ctx, ts, user, AuditLog.ACTION_SYSTEM_REPORT)
+
+                        if (config.slack.slackEmojiAdminChannel.isNotEmpty()) {
+                            logger.warn("${prop.id} has ${config.votes.downVoteThreshold} downvotes. Notify admins.")
+
+                            var body = when (prop.action) {
+                                Proposals.ACTION_ADD ->
+                                    ":heavy_plus_sign:`:${prop.emoji}:` "
+                                Proposals.ACTION_REMOVE ->
+                                    ":heavy_minus_sign:`:${prop.emoji}:` "
+                                else ->
+                                    ""
+                            }
+
+                            if (prop.alias) body += "(alias for `:${prop.cname}:` :${prop.cname}:) "
+
+                            body += "<${prop.permalink}|Thread> "
+                            if (tallyResult.block - tallyResult.unblock > 0) body += ":${Proposals.BLOCK}: "
+
+                            body += "${tallyResult.up}:${Proposals.UPVOTE}: ${tallyResult.down}:${Proposals.DOWNVOTE}: net: ${tallyResult.up - tallyResult.down}"
+                            if (tallyResult.up - tallyResult.down > config.votes.winBy) body += ":+1: "
```

Comment:
this is fine (not suggesting it be changed now), but if you wanted to make it more easily testable, you could pull some of this logic out into functions. e.g. you could have a TallyResult enum class that includes an emoji text property, with values like `APPROVED`, `DECLINED`, or whatever the enum results are that you think they should be. then this area could be something like:
`body.append(determineTallyResult(tallyResult.up, tallyResult.down, config.votes.winBy).emojiText)`